### PR TITLE
Improve Uploader

### LINKED ISSUE #674: Show a preview of directory contents in the upload window
The app should show a list of already-existing files in the upload window. This will provide additional context for where the user is in the directory hierarchy, reducing the complexity of the required mental model. It may also help the user decide if they want to upload the file or not (e.g. if they see that there will be a conflict).

I should point out that Dropbox and (IIRC) Google Drive do this.

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
fixes #674, #198

### CODE REVIEW

**Comment by strugee** [comment]
@jancborchardt did you ever file a separate issue? Seems like there's no reason to delay this PR.

**Comment by strugee** [comment]
Nevermind, that's #828.

**Comment by jancborchardt** [comment]
@davivel @masensio @rperezb review this please.

**Comment by masensio** [comment]
Hi @tobiasKaminsky,

I'm reviewing this PR, but the branch is outdated. I'll updated it with master branch :)

**Comment by masensio** [comment]
Branch updated

**Comment by davivel** [comment]
@AndyScherzinger , FYI: you can add comments directly on the 'guilty' lines of code. Go to the 'Files changed' tab, and click on the '+' icon that appears hovering the cursor on a line.

**Comment by davivel** [comment]
Delayed to 1.8.1

**Comment by MTRichards** [comment]
Sorry for the delay everyone, I know we have a backlog here and a few old PRs. We are trying to keep up with it all, but we are heads down in working on internal sharing and folder syncing at this time, and are on a tight timeline for a customer commitment. Just wanted people to know why things are delayed on our side, and will be for the next month or so.

File: src/com/owncloud/android/ui/activity/Uploader.java [hunk 1]
```diff
@@ -444,6 +485,27 @@ private void populateDirectoryList() {
...
+                                                                        false,
+                                                                        false,
+                                                                        false,
+                                                                        getStorageManager(),
+                                                                        getAccount(),
+                                                                        getApplicationContext()
+                                                                      );
+        synchFolderOp.execute(getAccount(), this, null, null);
+        
+        setSupportProgressBarIndeterminateVisibility(true);
```
Comment (AndyScherzinger) [inline src/com/owncloud/android/ui/activity/Uploader.java]: won't work anymore with the actual action bar and must be implemented as seen in FileDisplayActivity, e.g. in the onCreate method line 214 https://github.com/owncloud/android/blob/master/src/com/owncloud/android/ui/activity/FileDisplayActivity.java#L214

**Comment by tobiasKaminsky** [comment]
@MTRichards Thank you for the update on your side. :+1:

**Comment by AndyScherzinger** [comment]
@MTRichards like @tobiasKaminsky Thanks for the update from your side having such infos help a lot with my own planning :+1:

File: src/com/owncloud/android/ui/activity/Uploader.java (hunk 1 above)
Comment (tobiasKaminsky) [inline src/com/owncloud/android/ui/activity/Uploader.java]: Maybe you can have a look?
I tried to get mProgressBar via mProgressBar = (ProgressBar) findViewById(R.id.progressBar);
But it is not there.

Also the layout looks on a Nexus5 Emulator strange:
uploader

Comment (AndyScherzinger) [inline src/com/owncloud/android/ui/activity/Uploader.java]: Ah, the progressbar has been implemented on the FileDisplayActivity and Uploader is extending FileActivity, so no Progressbar there :(

You might have to add the progress bar to your uploader layout and look up the element in populateDirectoryList()

**Comment by AndyScherzinger** [comment]
Did you update to the latest master? Since the screenshot still shows the old iconset?!

**Comment by tobiasKaminsky** [comment]
Uh. No. Then everything should be ok with the view ;)

**Comment by AndyScherzinger** [comment]
True, except for the progressBar. That probably needs to be triggered when you get back to the FileDisplayActivity

File: src/com/owncloud/android/ui/activity/Uploader.java (hunk 1 above)
Comment (tobiasKaminsky) [inline src/com/owncloud/android/ui/activity/Uploader.java]: I think this is a little bit beyond my scope ;)
I hate these ui depentant things...

Comment (AndyScherzinger) [inline src/com/owncloud/android/ui/activity/Uploader.java]: It probably makes sense to not trigger the progress visualization and you should be fine then :)

**Comment by jancborchardt** [comment]
@MTRichards cool, awesome on the info! :+1: :)

**Comment by davivel** [comment]
With the current schedule there is no way we can have this done for https://github.com/owncloud/android/milestones/1.9-current .

Moving to next release.

cc @rperezb , @MTRichards , @cmonteroluque 

Notice that this solves a blue-ticket, #198.

**Comment by rperezb** [comment]
hey @tobiasKaminsky can you please update this pr with master? thx!

**Comment by tobiasKaminsky** [comment]
@rperezb I have updated the branch.

**Comment by tobiasKaminsky** [comment]
please see #1399 as this one is now a PR against master and not develop.

**Comment by jancborchardt** [comment]
@tobiasKaminsky so we can delete this branch as it’s outdated?

**Pull request closed without merging**